sys/include/isrpipe.h

```diff
@@ -54,10 +54,22 @@ typedef struct {
  */
 void isrpipe_init(isrpipe_t *isrpipe, uint8_t *buf, size_t bufsize);
 
+/**
+ * @brief   Put number of characters into the isrpipe's buffer
+ *
+ * @param[in]   isrpipe     isrpipe object to operate on
+ * @param[in]   buf         characters to add to isrpipe buffer
+ * @param[in]   n           number of characters to add from buf to isrpipe's buffer
+ *
+ * @returns     number of characters that could be added
+ * @returns     -1 if buffer was full
+ */
+int isrpipe_write(isrpipe_t *isrpipe, uint8_t *buf, size_t n);
```

Comment:
```suggestion
int isrpipe_write(isrpipe_t *isrpipe, const void *buf, size_t n);
```

Reply:
Isn't it better to keep consistency with `isrpipe_init` which also uses `uint8_t *buf` ?

Reply:
You can add arbitrary data here, no? The API doesn't have to care about what buffer is used to back the data structure

Reply:
Sure, but `ringbuffer` for instance even uses `const char *buf` so maybe it's worth a thought to remodel those values in general.

Reply:
It's just more convenient to use an API that takes a `void *` for arbitrary data than a `uint8_t *` - let the function do the cast if it needs to operate on bytes, not the caller.

Reply:
See also https://github.com/RIOT-OS/RIOT/issues/5497

Reply:
I don't have a strong opinion here, maybe just in favor of keeping things aligned, currently isrpipe is lying on top of `tsrb` which is using `uint8_t`, and all current users use `uint8_t`.

Reply:
Should be `const` still

Reply:
I added a `const` qualifier. I'm for keeping `uint8_t` as well, for uniformity.